Yoast/wordpress-seo
# Use right url for calculating

## Summary

This PR can be summarized in the following changelog entry:

* Fixes a bug where the internal links aren't recognized when the `site_url` is not the same as the `home_url`

## Relevant technical choices:

* I've used an upgrade routine to make sure everything has been counted again

## Test instructions

This PR can be tested by following these steps:
1. Have an installation (I've used acceptance.test) where all the WordPress files are located in a `/wp` directory and have in the database in the options table the option `site_url` set to `domainblablaba/wp.test` (In my situation: `acceptance.test/wp`). 
1. Checkout `trunk`
1. Make 2 posts and have one of them linking to the other one. ( example 1 links to example 2)
 In the internal link columns in the link overview the counts are wrong or aren't filled.
1. Checkout this branch and follow the steps 3.
1. The result of step 4 should be solved now.

* The upgrade routine can be triggered by setting the WordPress SEO version to 6.1 (https://github.com/Yoast/wordpress-seo/blob/trunk/wp-seo-main.php#L16) and in the database option (`wpseo`) the value should be something below 6.1 (like 5.9)

Fixes #8336
